### PR TITLE
feat: display deployments and nodes conditions in a table

### LINKED ISSUE #9121: Make Kubernetes resources conditions more readable
### Is your enhancement related to a problem? Please describe

The conditions are displayed this way:
conditions-before

### Describe the solution you'd like

Displayed in a table, as in OpenShift web ui for example:

conditions-after

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by feloy**
Signed-off-by: Philippe Martin <phmartin@redhat.com>

### What does this PR do?

Display the Deployments and Nodes condition in a table.
 
### Screenshot / video of UI

Deployments before

deploy-conditions-before

Deployments after

deploy-conditions-after

Nodes before

node-conditions-before

Nodes after

node-conditions-after

### What issues does this PR fix or reference?

Fixes #9121

### How to test this PR?

Check conditions in Deployments Details > Summary and Nodes Details > Summary

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/renderer/src/lib/kube/details/KubeDeploymentStatusArtifact.svelte [hunk 1]
```diff
@@ -23,26 +25,23 @@ export let artifact: V1DeploymentStatus | undefined;
...
+              <td>{condition.type}</td>
+              <td>{condition.status}</td>
+              <td>{humanizeDuration(moment().diff(condition.lastTransitionTime), { round: true, largest: 1 })}</td>
+              <td>{condition.reason}</td>
+              <td>{condition.message}</td>
+            </tr>
+          {/each}
+        </Table>
+      </td>
+    </tr>
```
Comment (benoitf) [inline packages/renderer/src/lib/kube/details/KubeDeploymentStatusArtifact.svelte]: todo: we have the exact same lines in the other file, so create a subcomponent .svelte

and then just call `<StatusTable={artifact.conditions} />`

Comment (feloy) [reply packages/renderer/src/lib/kube/details/KubeDeploymentStatusArtifact.svelte]: done

File: packages/renderer/src/lib/kube/details/ConditionsTable.spec.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+/**********************************************************************
+ * Copyright (C) 2023-2024 Red Hat, Inc.
```
Comment (benoitf) [inline packages/renderer/src/lib/kube/details/ConditionsTable.spec.ts]:
```suggestion
 * Copyright (C) 2024 Red Hat, Inc.
```

Comment (benoitf) [reply packages/renderer/src/lib/kube/details/ConditionsTable.spec.ts]: nitpick

**Review by benoitf — APPROVED** [review]

**Review by deboer-tim — APPROVED** [review]
Visually the DetailsTable gives a little too much padding around the conditions, but not worth fixing for now. Thanks, much better to show the data like this!

**Review by axel7083 — APPROVED** [review]
LGTM

**Pull request merged**